File: Utils/src/main/java/com/sheryv/util/logging/ConsoleUtils.java
package com.sheryv.util.logging;

import org.apache.commons.lang3.RegExUtils;
import org.apache.commons.lang3.StringUtils;

import java.util.HashMap;
import java.util.Map;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ConsoleUtils {
  private static final Pattern PATTERN_COLOR = Pattern.compile("(&\\d\\d?&)");
  
  private ConsoleUtils() {
  }
  
  public static String parseAndReplaceWithColors(String text) {
    Matcher matcher = PATTERN_COLOR.matcher(text);
    String result = text+"";
    while (matcher.find()) {
      String group = matcher.group();
      String substring = text.substring(matcher.start(), matcher.end());
      String code = group.substring(1, group.length() - 1);
      code = COLORS.get(code);
      code = code != null ? code : "";
      result = result.replace(substring, code);
    }
    return result;
//    return matcher.replaceAll(matchResult -> {
//      String group = matchResult.group();
//      String code = group.substring(1, group.length() - 1);
//      code = COLORS.get(code);
//      return code != null ? code : "";
//    });
  }
  
  private static final Map<String, String> COLORS = new HashMap<>();
  
  
  static final String RESET = "\033[0m";  // Text Reset
  
  static final String BLACK = "\033[0;30m";   // BLACK
  static final String RED = "\033[0;31m";     // RED
  static final String GREEN = "\033[0;32m";   // GREEN
  //   static final String YELLOW = "\033[0;33m";  // YELLOW
  static final String BLUE = "\033[0;34m";    // BLUE
  //   static final String PURPLE = "\033[0;35m";  // PURPLE
  static final String CYAN = "\033[0;36m";    // CYAN
  static final String WHITE = "\033[0;37m";   // WHITE
  
  
  static final String BLACK_BRIGHT = "\033[0;90m";  // BLACK
  static final String RED_BRIGHT = "\033[0;91m";    // RED
  static final String GREEN_BRIGHT = "\033[0;92m";  // GREEN
  static final String YELLOW_BRIGHT = "\033[0;93m"; // YELLOW
  static final String BLUE_BRIGHT = "\033[0;94m";   // BLUE
  static final String PURPLE_BRIGHT = "\033[0;95m"; // PURPLE
  static final String CYAN_BRIGHT = "\033[0;96m";   // CYAN
  static final String WHITE_BRIGHT = "\033[0;97m";  // WHITE
  
  static {
    COLORS.put("0", RESET);
    COLORS.put("1", GREEN);
    COLORS.put("2", CYAN);
    COLORS.put("3", RED);
    COLORS.put("4", BLUE);
    COLORS.put("5", WHITE);
    COLORS.put("6", BLACK);
  }
}

File: Utils/src/main/kotlin/com/sheryv/util/logging/ConsoleUtils.kt
package com.sheryv.util.logging

import java.util.regex.Pattern

object ConsoleUtils {
  private val PATTERN_COLOR = Pattern.compile("(&\\d\\d?&)")
  
  @JvmStatic
  fun parseAndReplaceWithColors(text: String): String {
    val matcher = PATTERN_COLOR.matcher(text)
    var result = text + ""
    while (matcher.find()) {
      val group = matcher.group()
      val substring = text.substring(matcher.start(), matcher.end())
      var code: String? = group.substring(1, group.length - 1)
      code = COLORS[code]
      code = code ?: ""
      result = result.replace(substring, code)
    }
    return result
    //    return matcher.replaceAll(matchResult -> {
//      String group = matchResult.group();
//      String code = group.substring(1, group.length() - 1);
//      code = COLORS.get(code);
//      return code != null ? code : "";
//    });
  }
  
  
  const val RESET = "\u001b[0m" // Text Reset
  
  const val DEFAULT = "\u001b[39m" // Text Reset
  const val DEFAULT_WITH_RESET = "\u001b[0;39m" // Text Reset
  const val BLACK = "\u001b[30m" // BLACK
  const val RED = "\u001b[31m" // RED
  const val GREEN = "\u001b[32m" // GREEN
  const val YELLOW = "\u001B[33m"  // YELLOW
  const val BLUE = "\u001b[34m" // BLUE
  const val PURPLE = "\u001b[35m"  // PURPLE
  const val CYAN = "\u001b[36m" // CYAN
  const val WHITE = "\u001b[37m" // WHITE
  const val BLACK_BRIGHT = "\u001b[90m" // BLACK
  const val RED_BRIGHT = "\u001b[91m" // RED
  const val GREEN_BRIGHT = "\u001b[92m" // GREEN
  const val YELLOW_BRIGHT = "\u001b[93m" // YELLOW
  const val BLUE_BRIGHT = "\u001b[94m" // BLUE
  const val PURPLE_BRIGHT = "\u001b[95m" // PURPLE
  const val CYAN_BRIGHT = "\u001b[96m" // CYAN
  const val WHITE_BRIGHT = "\u001b[97m" // WHITE
  
  fun addBold(colorCode: String) = "\u001B[1;" + colorCode.drop(2)
  
  fun addReset(colorCode: String) = "\u001B[0;" + colorCode.drop(2)
  
  private val COLORS: Map<String, String> = mapOf(
    "0" to RESET,
    "1" to GREEN,
    "2" to CYAN,
    "3" to RED,
    "4" to BLUE,
    "5" to WHITE,
    "6" to BLACK,
  )
  
  val COLORS_BY_NAMES: Map<String, String> = mapOf(
    "DEFAULT" to DEFAULT,
    "BLACK" to BLACK,
    "RED" to RED,
    "GREEN" to GREEN,
    "YELLOW" to YELLOW,
    "BLUE" to BLUE,
    "PURPLE" to PURPLE,
    "CYAN" to CYAN,
    "WHITE" to WHITE,
    "BLACK_BRIGHT" to BLACK_BRIGHT,
    "RED_BRIGHT" to RED_BRIGHT,
    "GREEN_BRIGHT" to GREEN_BRIGHT,
    "YELLOW_BRIGHT" to YELLOW_BRIGHT,
    "BLUE_BRIGHT" to BLUE_BRIGHT,
    "PURPLE_BRIGHT" to PURPLE_BRIGHT,
    "CYAN_BRIGHT" to CYAN_BRIGHT,
    "WHITE_BRIGHT" to WHITE_BRIGHT,
  )
  
}
